### PR TITLE
1596: implement marker configuration in Standalone Map

### PR DESCRIPTION
**Pull request by adnauseum**
**Before**:

Screenshot 2023-04-25 at 9 45 50 AM

Here are the designs:

image
image

**After**:

https://user-images.githubusercontent.com/24446573/235502941-3268e476-2802-4dfc-a321-f9dfa08029e2.mp4

image

### CODE REVIEW

**Comment by dmfalke** [comment]
Screencast looks great! I will review the code as soon as I can (probably tonight).

File: packages/libs/eda/src/lib/map/analysis/MarkerConfiguration/BarPlotConfigurationMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  starredVariables,
+  toggleStarredVariable,
+  configuration,
+}: Props) {
+  function handleInputVariablesOnChange(selection: VariablesByInputName) {
+    if (!selection.overlay) {
+      console.error(
+        `Expected overlay to defined but got ${typeof selection.overlay}`
+      );
+      return;
```
Comment (adnauseum) [inline packages/libs/eda/src/lib/map/analysis/MarkerConfiguration/BarPlotConfigurationMenu.tsx:43]: image

☝️ I imagine that we want to update this component to take a prop to hide the clear button. If not, I suppose we could select the study's default overlay variable.

**Review by dmfalke — CHANGES_REQUESTED** [review]
I think the code looks great. I have a couple of requested changes:

1. Make the "clear" button optional for `InputVariables`
2. Change the name of `DonutConfigurationMenu` to `DonutMarkerConfigurationMenu`, and `BarPlotConfigurationMenu` to `BarPlotMarkerConfigurationMenu`

File: packages/libs/eda/src/lib/core/components/VariableLink.tsx [hunk 2]
```diff
@@ -33,14 +33,8 @@ export interface Props<S = unknown> extends Omit<LinkProps<S>, 'to'> {
...
-    const {
-      entityId,
-      disabled,
-      variableId,
-      linkConfig,
-      style,
-      ...rest
-    } = props;
+    const { entityId, disabled, variableId, linkConfig, style, ...rest } =
+      props;
```
[the remark this answers is not in the document]
Comment (adnauseum) [reply packages/libs/eda/src/lib/core/components/VariableLink.tsx:37]: > Result of a formatting tool?
Yep, I was poking around that file, saved (muscle memory), and the new prettier config was applied 😄 

> Change the name of DonutConfigurationMenu to DonutMarkerConfigurationMenu, and BarPlotConfigurationMenu to BarPlotMarkerConfigurationMenu

Done in: 
* https://github.com/VEuPathDB/web-monorepo/pull/111/commits/4fd70ed3031a43ec79231995cca8461aa460266a
* https://github.com/VEuPathDB/web-monorepo/pull/111/commits/4ca19ee628642343271b9841588f2015163a4f05

> Make the "clear" button optional for InputVariables

Added in: https://github.com/VEuPathDB/web-monorepo/pull/111/commits/92e2d79a43ad4940161a3db3b9b162cc1e884744

🫡

**Pull request merged**